src/main/java/org/craftercms/studio/api/v2/content/LifeCycleContent.java

```diff
@@ -0,0 +1,251 @@
+ */
+public class LifeCycleContent implements AutoCloseable {
+
+	private final String repoPath;
+	private final LifeCycleOperation operation;
+	private final String contentType;
+	private final Map<String, ContentLifeCycleItem> items;
+
+	/**
+	 * Constructor for creating a new LifecycleContent object.
+	 *
+	 * @param repoPath        the path to the content item in the repository
+	 * @param sourcePath      source path for move operations
+	 * @param contentType     the content type of the item
+	 * @param contentProvider provider to get the content from
+	 * @param operation       the life cycle operation to be performed
+	 */
+	public LifeCycleContent(String repoPath, String sourcePath, String contentType, LifeCycleContentProvider contentProvider, LifeCycleOperation operation) {
+		this.items = new HashMap<>();
+		this.repoPath = repoPath;
+		this.operation = operation;
+		this.contentType = contentType;
+		this.items.put(repoPath, new ContentLifeCycleItem(repoPath, sourcePath, contentProvider));
+	}
+
+	/**
+	 * Constructor for creating a new LifecycleContent object.
+	 *
+	 * @param repoPath        the path to the content item in the repository
+	 * @param contentType     the content type of the item
+	 * @param contentProvider provider to get the content from
+	 * @param operation       the life cycle operation to be performed
+	 */
+	public LifeCycleContent(String repoPath, String contentType, LifeCycleContentProvider contentProvider, LifeCycleOperation operation) {
+		this(repoPath, null, contentType, contentProvider, operation);
+	}
+
+	/**
+	 * Add a new content item to the life cycle operation.
+	 *
+	 * @param path    the path to the content item
+	 * @param content InputStream of the content item
+	 * @throws IOException if an error occurs while reading the stream or storing the content
+	 */
+	public void write(String path, InputStream content) throws IOException {
+		Path filePath = createTempFile(path, content);
+		write(path, filePath);
+	}
+
+	/**
+	 * Write the content to the given path in the repository.
+	 *
+	 * @param path     the path to write the content to
+	 * @param document the Document to write
+	 * @throws IOException if an error occurs while writing the content
+	 */
+	public void write(String path, Document document) throws IOException {
+		Path filePath = createTempFile(path, document);
+		write(path, filePath);
+	}
+
+	/**
+	 * Write the content to the given path in the repository.
+	 *
+	 * @param path     the path to write the content to
+	 * @param filePath the path containing the content to write
+	 */
+	public void write(String path, Path filePath) {
+		String normalizedPath = FilenameUtils.normalize(path);
+		// Remove the temporary file if it exists
+		exclude(normalizedPath);
+		// Add a new entry with amended=<path is the same as the original repoPath>
+		this.items.put(normalizedPath, new ContentLifeCycleItem(normalizedPath, ofPath(() -> filePath), repoPath.equals(normalizedPath)));
```

Comment:
Is the rabbit right?

Reply:
No. Please see my response to its comment